src/main/java/com/dataguise/saas/service/OverviewService.java

```diff
@@ -0,0 +1,316 @@
+package com.dataguise.saas.service;
+
+import com.dataguise.saas.dto.*;
+import com.dataguise.saas.exception.BadGatewayException;
+import com.dataguise.saas.exception.BadRequestException;
+import com.dataguise.saas.exception.InternalServerException;
+import com.dg.saas.orch.client.DgSecureRestDriver;
+import com.dg.saas.orch.exception.DgDrvException;
+import com.dg.saas.orch.models.structures.CustomObject;
+import com.dg.saas.orch.models.structures.DrillDownDetailedResponse;
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import com.dg.saas.orch.models.structures.Modules;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import io.swagger.client.model.*;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * The Overview service
+ */
+@Service
+public class OverviewService {
+
+    private static Logger logger = Logger.getLogger(OverviewService.class);
+
+    @Autowired
+    private DgSecureRestDriver restDriver;
+
+    @Autowired
+    private ConnectionService connectionService;
+
+    @Autowired
+    private ProtectionTaskService protectionTaskService;
+
+    @Autowired
+    SourceHierarchyExplorerService sourceHierarchyExplorerServiceService;
+
+    /**
+     * Gets Source Count of cloud and on premise sources.
+     *
+     * @return SourceCount
+     */
+    public ResponseEntity getSourceCount() throws BadGatewayException {
+        SourceCountDTO sourceCount = new SourceCountDTO();
+        Integer cloud = 0;
+        Integer on_premise = 0;
```

Comment:
**Reporter**: PMD    
    **Rule**: LocalVariableNamingConventions
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/OverviewService.java L57
    
    The local variable name &#39;on_premise&#39; doesn&#39;t match &#39;[a-z][a-zA-Z0-9]*&#39;&#10;&#10;&#10;Code Style https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#localvariablenamingconventions
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-29386644>*